core/jvm/src/main/scala/zio/metrics/jvm/BufferPools.scala

```diff
@@ -0,0 +1,45 @@
+package zio.metrics.jvm
+
+import zio._
+
+import java.lang.management.{BufferPoolMXBean, ManagementFactory}
+
+object BufferPools extends JvmMetrics {
+
+  /** Used bytes of a given JVM buffer pool. */
+  private def bufferPoolUsedBytes(pool: String): ZIOMetric.Gauge[Long] =
+    ZIOMetric.setGaugeWith("jvm_buffer_pool_used_bytes", MetricLabel("pool", pool))(_.toDouble)
+
+  /** Bytes capacity of a given JVM buffer pool. */
+  private def bufferPoolCapacityBytes(pool: String): ZIOMetric.Gauge[Long] =
+    ZIOMetric.setGaugeWith("jvm_buffer_pool_capacity_bytes", MetricLabel("pool", pool))(_.toDouble)
+
+  /** Used buffers of a given JVM buffer pool. */
+  private def bufferPoolUsedBuffers(pool: String): ZIOMetric.Gauge[Long] =
+    ZIOMetric.setGaugeWith("jvm_buffer_pool_used_buffers", MetricLabel("pool", pool))(_.toDouble)
+
+  private def reportBufferPoolMetrics(
+    bufferPoolMXBeans: List[BufferPoolMXBean]
+  ): ZIO[Any, Throwable, Unit] =
+    ZIO.foreachParDiscard(bufferPoolMXBeans) { bufferPoolMXBean =>
+      for {
+        name <- Task(bufferPoolMXBean.getName)
+        _    <- Task(bufferPoolMXBean.getMemoryUsed) @@ bufferPoolUsedBytes(name)
```

Comment:
Not related to this PR but you could imagine there being an `unsafe` method to update a metric so this whole thing could take place within one `ZIO.succeed` block.

Reply:
That would probably also help in some migration situations allowing you to (temporarily) not convert something to an effect that deep inside called some non-ZIO metric API to report a metric.